### DISCUSSION TITLE
Enabling Lid Switch and sleep

### DISCUSSION THREAD

**Discussion by smedir** [jglathe/linux_ms_dev_kit#21] [Q&A]
Booting Debian Trixie with cross compiled rc-4 using your config.

How do you enable the lid switch? I see you mentioned in another thread that it is working.

And is your system able to sleep or hibernate successfully?

**Comment by jglathe** [comment]
I looked into the source of the CRD for clues, and copied the [relevant sections](https://github.com/anonymix007/linux-x1e/commit/88becf3b5b260881f93a333ab474899eb8314007). And was lucky on the HP X14 (works). on #aarch64-laptops this is discussed, looks like it may be different there. Something about EC being part of it.

**Reply by smedir** [reply]
I tried the patch for EC.

Applied 3 patch set from https://lore.kernel.org/all/20240927185345.3680-1-maccraft123mc@gmail.com/ to your 6.12.rc3

In .config enabled EC_LENOVO_YOGA_SLIM7X=y

rebuild kernel pkg and install.

Copy new dtb file to boot.

Boot into new kernel - mute key working, lid switch and sleep not working.

modprobe lenovo_yoga_slim7x - seems it is built as a kernel module.

Mute key works. Lid switch still does not work :disappointed: 

Do I have to apply your gpio patch to the dts as well?

If anybody has any ideas, let me know.

**Reply by smedir** [reply]
Actually I take that back.

The laptop suspends correctly and the power button light starts flashing on suspend.

However it does not suspend on closing the lid.
